src/docs/sphinx/pythonTools/mesh_doctor.rst

```diff
@@ -122,3 +122,60 @@ It will also verify that the ``VTK_POLYHEDRON`` cells can effectively get conver
 
 .. command-output:: python mesh_doctor.py supported_elements --help
    :cwd: ../../../coreComponents/python/modules/geosx_mesh_doctor
+
+``Using mesh_doctor in paraview``
+""""""""""""""""""""""""""""""""""
+To use ``mesh_doctor`` in Paraview as a python programmable filter, a python package install is required first in Paraview python resolved
+path. Paraview is storing its python ressources under its *lib/pythonX.X* depending on the paraview version, *e.g* Paraview 5.11 is working
+with python 3.9. As a results the following command will install ``mesh_doctor`` package into Paraview resolved path.
+
+.. command-output:: python3 -m pip install --index-url https://test.pypi.org/simple/ --no-deps --upgrade --target /path/to/Paraview/lib/python3.9/ mesh_doctor
+
+.. note::
+    ``pip`` is installing the ``mesh_doctor`` package from the test.pypi repo, which is intended to test package deployment.
+    Once stabilized and ``mesh_doctor`` uploaded onto the main package repo, this should be dropped out.
+
+Once the installation done, the */path/to/Paraview/lib/pythonX.X* should holds ``mesh_doctor`` package content, *i.e.* ``checks`` and ``parsing``.
+Then launching ``Paraview`` and loading our *mesh.vtu*, as an example, we will design a *Programmable python filter* relying on *element_volumes* from
+``mesh_doctor``. Add such a filter pipelined after the mesh reader, in the script section paste the following,
+
+.. code-block:: python
+    :linenos:
+
+    input0 = inputs[0]
+    tol = 1.2e-6
+
+    from checks import element_volumes
+    import vtk
+    #
+    res = element_volumes.__check(inputs[0].VTKObject, element_volumes.Options(tol))
+    #print(res)
+    ids = vtk.vtkIdTypeArray()
+    ids.SetNumberOfComponents(1)
+    for val in res.element_volumes:
+     ids.InsertNextValue(val[0])
+    #
+    selectionNode = vtk.vtkSelectionNode()
+    selectionNode.SetFieldType(vtk.vtkSelectionNode.CELL)
+    selectionNode.SetContentType(vtk.vtkSelectionNode.INDICES)
+    selectionNode.SetSelectionList(ids)
+    selection = vtk.vtkSelection()
+    selection.AddNode(selectionNode)
+    extracted = vtk.vtkExtractSelection()
+    extracted.SetInputDataObject(0, inputs[0].VTKObject)
+    extracted.SetInputData(1, selection)
+    extracted.Update()
+    print("There are {} cells under {} m3 vol".format(extracted.GetOutput().GetNumberOfCells(), tol))
+    output.ShallowCopy(extracted.GetOutput())
+
+Here we rely on ``pyvtk`` interface more than on Paraview adaptation, for legacy and reusability reasons. This is the reason
```

Comment:
Do you mean that we cannot access `vtkSelectionNode` from within `paraview`?
We may want to find a way to detect that we're in the "paraview plugin env" to rely on the paraview vtk not to risk any conflict.
But the `requirements.txt` part remains. However, we'll move forward!

Reply:
That is exactly that. As I tried to launched it having `from paraview import vtk` instead of `import vtk`, I bumped into an error saying that there was no `vtkSelectionNode`. As _Paraview_ is able to do the operation, I guess this is just that this is not exposed but wrapped. But there must be a way through _Paraview_, it is just a question of 1. I started from a pyvtk script, which I found more versatile but not Paraview compativle  and 2. we might want to rewrite the script in a more _Paraview_ way then.